src/bin/pg_autoctl/cli_config.c

```diff
@@ -650,23 +801,12 @@ cli_keeper_config_set(int argc, char **argv)
 		/* we print out the value that we parsed, as a double-check */
 		char value[BUFSIZE];
 
-		if (!keeper_config_set_setting(&config,
-									   argv[0],
-									   argv[1]))
+		if (!cli_keeper_config_validate_and_commit(&config, argv[0], argv[1]))
```

Comment:
Are we running the new validation code for any setting that can be set through `pg_autoctl config set` now, rather than only the replication password?

Reply:
Apologies. We forgot to guard the replication validation check like this:

```diff
-	if (!cli_keeper_config_validate_replication(config, &pgsql))
+	if (streq(key, "replication.password") && !cli_keeper_config_validate_replication(config, &pgsql))
	{
```
We will correct this.

Reply:
I remember why we did this -- the other configuration settings seemed to be validated as a whole, and so any co-dependencies between them (say, hostname and port) would be handled. We were worried that setting some _other_ configuration could effectively invalidate the password (especially given the reliance on HBA), and that we wouldn't end up discovering the failure until it was already committed. Is the cost of running the validation too high?

Reply:
Good point, we should re-compute the primary_conninfo connection string for every config setting that would impact it. In practice it's mostly SSL and password. 

I'm not so sure about when a node's hostname and port are edited, we could also re-check although I suppose it comes with network updates that might have required a system / pg_autoctl restart already, and the change needs to be done on the monitor and then all other nodes in cascade, which we already take into account. Making sure that we test the change locally at the right time (once the HBA have been edited everywhere) is going to be rather complex, if not impossible. I would not sweat over it.

Doing the primary conninfo change for all parameters is too much in my view, yes.